theano/tensor/signal/pool.py

```diff
@@ -255,6 +255,20 @@ def pool_3d(input, ws=None, ignore_border=None, stride=None, pad=(0, 0, 0),
     return output
 
 
+def spp_pooling(image, out_dimension, image_shape):
+
+    input_size = image_shape[2:]
+    pooled_data_list = []
+    for pool_dim in out_dimension:
+        pool_size = tuple((i + pool_dim - 1) // pool_dim for i in input_size)
+        stride_size = tuple((i // pool_dim) for i in input_size)
+        pooled_part = pool_2d(image, pool_size, ignore_border=True, st=stride_size, padding=(0, 0), mode='max')
+        pooled_part = pooled_part.flatten(3)
+        pooled_data_list.append(pooled_part)
+
+    return tensor.concatenate(pooled_data_list, axis=2)
```

Comment:
I think you only need this implementation. I don't see why you need the one in dnn. The optimizer should move it completly to the GPU. If I'm missing something, tell me.

It was maybe needed in the past before we had our new pool interface, but I don't think it is needed anymore.

For the name, I'm not sure what we should use. Do you know what they use in other software like torch, tensorflow or caffe for that name?

This will also need a docstring.

I looked at the paper very rapidly, and I don't think a new op is useful for this.

I think it would be useful to put a default of None for image_shape and if it is NOne, do: `image_shape = image.shape`. This now should work, but didn't worked in the past and probably why it was in lasagne dnn interface and not normal pool interface.